### PR TITLE
homework react week3 uploaded

### PR DESCRIPTION
**Pull request by amjadiftikhar**

### CODE REVIEW

File: react/week3/todo-app/src/components/addForm.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	const [state, setState] = useState({
+		description: "",
+		deadline: ""
+	});
+
+	const handleTodo = event => {
+		event.preventDefault();
+		if (!state) return;
+		addNewTodo(state);
+		setState("");
```
Comment (Hatko) [inline react/week3/todo-app/src/components/addForm.js]: Your state during initialization is an object (lines 4-7), here you reset it to an empty string

File: react/week3/todo-app/src/components/addTimer.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+function Timer() {
+	const [count, setCount] = useState(0);
+
+	useEffect(() => {
+		const counter = setTimeout(() => {
+			setCount(prev => prev + 1);
+		}, 1000);
+		return () => clearTimeout(counter);
+	});
```
Comment (Hatko) [inline react/week3/todo-app/src/components/addTimer.js]: There is a small issue I described here:

https://github.com/cecastosic/hyf-homework/pull/13#discussion_r379034665

File: react/week3/todo-app/src/components/addTodo.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import Timer from "./addTimer";
+import AddForm from "./addForm";
+
+function AddTodo() {
+	const [todos, setTodos] = useState([]);
+	useEffect(() => {
+		const getData = async () => {
+			const todoArray = await fetch(
+				"https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+			).then(response => response.json());
```
Comment (Hatko) [inline react/week3/todo-app/src/components/addTodo.js]: Using async/await is great 👍

Could be even better if you try to get rid of then here and have only await statements

File: react/week3/todo-app/src/components/addTodo.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			setTodos(todoArray);
+			// console.log("This is from todoArray", todoArray);
+		};
+		getData();
+	}, []);
+
+	const addNewTodo = description => {
```
Comment (Hatko) [inline react/week3/todo-app/src/components/addTodo.js]: `description` is a bit misleading name. You have an object with `todo` here, as I understand 🙂

File: react/week3/todo-app/src/components/addTodo.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	const removeTodo = id => {
+		console.log(id);
+		const newFilteredTodos = [...todos];
+		newFilteredTodos.splice(id, 1);
+		setTodos(newFilteredTodos);
+	};
+
+	const handleCheckBox = id => {
+		const newTodos = [...todos];
+		newTodos[id].isCompleted = !newTodos[id].isCompleted;
```
Comment (Hatko) [inline react/week3/todo-app/src/components/addTodo.js]: The state is modified directly here, which should be avoided. Try to fix it, and if you have any questions - feel free to drop me a message

File: react/week3/todo-app/src/components/addForm.js (hunk 1 above)
Comment (amjadiftikhar) [reply react/week3/todo-app/src/components/addForm.js]: Yes I got the point thats why it was complaining that controlled input getting converted to uncontrolled.

File: react/week3/todo-app/src/components/addTodo.js (hunk 3 above)
Comment (amjadiftikhar) [reply react/week3/todo-app/src/components/addTodo.js]: Actually, I did not get this that how we can get rid of 'then' here ?

File: react/week3/todo-app/src/components/addTodo.js (hunk 5 above)
Comment (amjadiftikhar) [reply react/week3/todo-app/src/components/addTodo.js]: Thank you for you feedback. As I tried my best to resolve it but not sure as I am getting array from API and object does not have isCompleted property but as I am continuing with the last weeks assignment where we had a static array of object with isCompleted property. Now I am bit confused that how can we handle this situation where we are fetching data from API. Your input will be highly appreciated regarding this. Thank you

**Pull request merged**